pkg/eval/str/str.go

```diff
@@ -92,6 +95,62 @@ import (
 // ▶ $false
 // ```
 
+//elvdoc:fn from-codepoints
+//
+// ```elvish
+// str:from-codepoints $number...
+// ```
+//
+// Outputs a string consisting of the given Unicode codepoints. Example:
+//
+// ```elvish-transcript
+// ~> str:from-codepoints 0x61
+// ▶ a
+// ~> str:from-codepoints 0x4f60 0x597d
+// ▶ 你好
+// ```
+//
+// @cf str:to-codepoints
+
+func fromCodepoints(nums ...int) (string, error) {
+	var b bytes.Buffer
+	for _, num := range nums {
+		if !utf8.ValidRune(rune(num)) {
+			return "", fmt.Errorf("invalid codepoint: %d", num)
+		}
+		b.WriteRune(rune(num))
+	}
+	return b.String(), nil
+}
+
+//elvdoc:fn from-utf8-bytes
+//
+// ```elvish
+// str:from-from-utf8-bytes $number...
+// ```
+//
+// Outputs a string consisting of the given Unicode bytes. Example:
+//
+// ```elvish-transcript
+// ~> str:from-utf8-bytes 0x61
+// ▶ a
+// ~> str:from-utf8-bytes 0xe4 0xbd 0xa0 0xe5 0xa5 0xbd
+// ▶ 你好
+// ```
+//
+// @cf str:to-utf8-bytes
+
+func fromUtf8Bytes(nums ...int) (string, error) {
+	var b bytes.Buffer
+	for _, num := range nums {
+		b.WriteByte(byte(num))
```

Comment:
Check if num is within the range of byte, and return an [OutOfRange](https://github.com/elves/elvish/blob/9858d5a5455ab157934d84b118a4d63c45dbfcd4/pkg/eval/errs/errs.go#L9) error if it isn't. Also add a test for this.

Reply:
Done. I used int(byte(num)) = num and int(rune(num)) num for the overflow check, not sure if your would preferred `<` and `>` checks for the range boundaries.

Reply:
Normally I wouldn't have an opinion, but since you need to spell out the range in the error anyway I'd prefer using comparison operators.